isofit/utils/apply_oe.py

```diff
@@ -463,6 +463,23 @@ def apply_oe(args):
                     output=outp,
                     chunksize=CHUNKSIZE,
                     flag=-9999,
+                    reducer=reducers.band_mean,
+                    n_cores=args.n_cores,
+                    loglevel=args.logging_level,
+                    logfile=args.log_file,
+                )
+
+        # Extract superpixels class
+        if paths.surface_class_file:
+            if not exists(paths.subs_class_path):
+                logging.info("Extracting " + paths.subs_class_path)
+                extractions(
+                    inputfile=paths.surface_class_file,
+                    labels=paths.lbl_working_path,
+                    output=paths.subs_class_path,
+                    chunksize=CHUNKSIZE,
+                    flag=-9999,
+                    reducer=reducers.class_priority,
```

Comment:
To generate surface-class assignments for superpixels, I use the extractions function. Now expanded to handle general reduction functions to move between fine- and coarse pixel resolution.